src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -1,63 +1,654 @@
+
+            _writer.Append($"public {(isAsync ? "async " : string.Empty)}{(isOverride ? "override " : string.Empty)}{returnType} {CreateMethodName(syncMethodName, isAsync)}(");
+            foreach (var parameter in parameters)
+            {
+                _writer.WriteParameter(parameter);
             }
+            _writer.Append($"{typeof(CancellationToken)} {cancellationTokenParameter} = default)");
+            using var _ = _writer.Scope();
+            WriteDiagnosticScope(_writer, new Diagnostic($"{_resourceContainer.Type.Name}.{syncMethodName}"), ClientDiagnosticsField, writer =>
+            {
+                _writer.WriteParameterNullChecks(parameters.ToList());
+                inner(_writer);
+            });
         }
 
-        private void WriteContainerCtors(CodeWriter writer, ResourceContainer resourceContainer)
+        /// <summary>
+        /// Builds the mapping between resource operations in Container class and that in RestOperations class.
+        /// For example `DedicatedHostRestClient.CreateOrUpdate()`
+        /// | resourceGroupName      | hostGroupName    | hostName | parameters |
+        /// | ---------------------- | ---------------- | -------- | ---------- |
+        /// | "Id.ResourceGroupName" | "Id.Parent.Name" | hostName | parameters |
+        /// </summary>
+        /// <param name="method">Represents a method in RestOperations class.</param>
+        private IEnumerable<ParameterMapping> BuildParameterMapping(RestClientMethod method)
         {
-            writer.WriteXmlDocumentationSummary($"Initializes a new instance of {resourceContainer.Type.Name} for mocking.");
-            using (writer.Scope($"protected {resourceContainer.Type.Name:D}()"))
+            var parameterMapping = new List<ParameterMapping>();
+            var dotParent = ".Parent";
+            var parentNameStack = new Stack<string>();
+
+            // loop through parameters of REST client method, map the leading string-like parameters to
+            // Id.ResourceGroupName, Id.ResourceGroupName.Parent.Name, Id.ResourceGroupName.Parent.Parent.Name...
+            // special case: type is enum and you can convert string to it (model-as-string), we should handle it as string
+            // special case 2: in paging scenarios, `nextLink` needs to be handled specially, so here we just ignore it
+            foreach (var parameter in method.Parameters)
             {
+                bool passThru = true;
+                string valueExpression = string.Empty;
+                if (string.Equals(parameter.Name, "nextLink", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    continue;
+                }
+                else if (IsStringLike(parameter.Type) && IsMandatory(parameter))
+                {
+                    passThru = false;
+                    if (string.Equals(parameter.Name, "resourceGroupName", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Why are only checking `resourceGroupName` here? It could be subscriptionId, location ..etc